Smt/Commands.lean

```diff
@@ -31,52 +38,40 @@ def defNatSub : Command :=
     (`"ite" • (`"<" • `"x" • `"y") • ``"0" • (`"-" • `"x" • `"y"))
     false
 
-def sortEndsWithNat : Term → Bool
-  | arrowT _ t    => sortEndsWithNat t
-  | symbolT "Nat" => true
-  | _             => false
-
-def natAssertBody (t : Term) : Term :=
-  mkApp2 (symbolT ">=") t (literalT "0")
+open ToSexp in
+protected def toSexp : Command → Sexp
+  | .setLogic l                   => sexp!{(set-logic {l})}
+  | .setOption k v                => sexp!{(set-option {s!":{k}"} {v})}
+  | .assert val                   => sexp!{(assert {val})}
+  | .declare nm st@(arrowT ..)    =>
+    let sts := arrowToList st
+    sexp!{(declare-fun {quoteSymbol nm} (...{sts.init.map toSexp}) {sts.getLast!})}
+  | .declare nm st                => sexp!{(declare-const {quoteSymbol nm} {st})}
+  | .declareSort nm arity         =>
+    sexp!{(declare-sort {quoteSymbol nm} {String.mk (Nat.toDigits 10 arity)})}
+  | .defineSort nm ps tm          =>
+    sexp!{(define-sort {quoteSymbol nm} (...{ps.map toSexp}) {tm})}
+  | .defineFun nm [] cod tm _     =>
+    sexp!{(define-const {quoteSymbol nm} {cod} { tm})}
```

Comment:
`define-const` doesn't exist in the SMT-LIB [standard](https://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.6-r2021-05-12.pdf), is it supported by CVC? In any case I'd go with `define-fun` so that all solvers work.

Reply:
It's supported by both `cvc5` and `z3`. Don't know about the others (probably support it). Good point though. I removed it.